### PR TITLE
doc: update NCS style guide

### PR DESCRIPTION
**Pull request by greg-fer**
Added new sections. Reorganized RST part.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/doc_styleguide.rst [hunk 1]
```diff
@@ -15,33 +15,91 @@ RST style guide
...
++------------------------+------------------------+---------------------------------------------------------+
+| Markup                 | Example                | Usage criteria                                          |
++========================+========================+=========================================================+
+| ``*emphasis*``         | *relaying*             | Emphasis or new terms.                                  |
++------------------------+------------------------+---------------------------------------------------------+
+| ````code````           | ``west update``        | Code within text, including file paths and names.       |
++------------------------+------------------------+---------------------------------------------------------+
+| ``**PCB**``            | **SW3**                | PCB names.                                              |
++------------------------+------------------------+---------------------------------------------------------+
+| ``:guilabel:`button``` | :guilabel:`Cancel`     | Clickable UI buttons.                                   |
```
Comment (ru-fu) [inline doc/nrf/doc_styleguide.rst]: Only buttons? I thought we're using it for all GUI elements (see for example https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/gs_ins_windows.html#setting-up-the-ses-environment ).

File: doc/nrf/doc_styleguide.rst [hunk 2]
```diff
@@ -15,33 +15,91 @@ RST style guide
...
++------------------------+------------------------+---------------------------------------------------------+
 
-It is also possible to define more than one default link text for a link, which can be useful if you frequently need a different link text::
+Do not use the following markup: ```..```.
```
Comment (ru-fu) [inline doc/nrf/doc_styleguide.rst]:
```suggestion
Do not use the following markup for italics: ```..```.
```

File: doc/nrf/doc_styleguide.rst [hunk 3]
```diff
@@ -15,33 +15,91 @@ RST style guide
...
 
-   .. _`Link text one`:
-   .. _`Link text two`: http://..
+For readability reasons, do not use any highlighting for titles of other pages or headings when they are mentioned in the text.
```
Comment (ru-fu) [inline doc/nrf/doc_styleguide.rst]: I don't agree with that - if we have to link to a Zephyr section that does not have an ID to link to, we use standard citing: "See *Section name* in :ref:`link`"
I think what you wanted to state here is that we shouldn't use highlighting for library names as used in the code - which means that if we get input like "The ``bl_crypto`` library ..." we change it to "The bootloader crypto library ...". At least that's what I wanted to know when we discussed this. ;)

File: doc/nrf/doc_styleguide.rst [hunk 4]
```diff
@@ -15,33 +15,91 @@ RST style guide
...
+For example, do not use emphasis to make the names of libraries stand out.
+Consider linking to these pages instead.
```
Comment (ru-fu) [inline doc/nrf/doc_styleguide.rst]: We should ALWAYS link if we can, not only consider it.

File: doc/nrf/doc_styleguide.rst [hunk 5]
```diff
@@ -120,6 +202,43 @@ To link directly to a doxygen reference from RST, use the following Breathe doma
 
       :cpp:enumerator:`ENUM_VALUE <DOXYGEN_GROUP::ENUM_VALUE>`
 
+PCB names
+=========
+
+When referring to specific Printed Circuit Board elements, use capitalization.
+For example, *Button 1*, *Switch 3*, *LED 1*.
```
Comment (ru-fu) [inline doc/nrf/doc_styleguide.rst]: Isn't it confusing to use italics here?

File: doc/nrf/doc_styleguide.rst [hunk 6]
```diff
@@ -120,6 +202,43 @@ To link directly to a doxygen reference from RST, use the following Breathe doma
...
+If you provide the code of a specific PCB element, write it in bold: **SW3**.
+
+If you are referring to a generic PCB element, do *not* use capitalization: *button*, *switch*, *LEDs*.
+
+
+Tables
+======
+
+Follow Zephyr's :ref:`zephyr:doc_guidelines` for tables.
+However, do not add table titles to describe the table.
```
Comment (ru-fu) [inline doc/nrf/doc_styleguide.rst]:
```suggestion
Do not add table titles to describe the table.
```
With the "however" it sounds like we're diverting from Zephyr's guidelines - but we aren't. Zephyr doesn't say anything about table captions, their style is just as inconsistent as ours. ;)

File: doc/nrf/doc_styleguide.rst [hunk 7]
```diff
@@ -120,6 +202,43 @@ To link directly to a doxygen reference from RST, use the following Breathe doma
...
+        :header-rows: 1
+
+        * - Heading 1
+          - Heading 2
+          - Heading 3
+        * - body row 1, column 1
+          - body row 1, column 2
+          - body row 1, column 3
+
+This caption guideline applies to both grid and list tables.
```
Comment (ru-fu) [inline doc/nrf/doc_styleguide.rst]: Do we need this? There are several other tables types, and they are not forbidden to use. And the caption guideline applies to all of them.

File: doc/nrf/doc_styleguide.rst [hunk 8]
```diff
@@ -15,33 +15,91 @@ RST style guide
...
+Keep titles and headings as short and to the point as possible.
+
+Titles
+------
+
+Mention the module name in the title and within the RST tag of the page, which is defined in the first line of the RST file.
+For example:
+
+.. code-block:: python
+   :caption: Example of page tag and title
```
[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/doc_styleguide.rst]: I would personally use it. Let's discuss offline. For the time being, I'll add missing captions (or change them to `none`).

File: doc/nrf/doc_styleguide.rst (hunk 4 above)
Comment (greg-fer) [reply doc/nrf/doc_styleguide.rst]: What about multiple links to the same library? Should we link only on first occurrence?

Comment (ru-fu) [reply doc/nrf/doc_styleguide.rst]: First occurrence and any explicit pointer to it - so if we're talking about an external library, for example, we should have another link when we point out that the library must be enabled.

File: doc/nrf/doc_styleguide.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (ru-fu) [reply doc/nrf/doc_styleguide.rst]: Why would we treat code-blocks different from tables?

[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/doc_styleguide.rst]: Depending on the coding language, you can have different highlighting in the code block. I would use it for this reason.

File: doc/nrf/doc_styleguide.rst [hunk 9]
```diff
@@ -15,33 +15,91 @@ RST style guide
...
+
+    1 .. _nfc_tag_reader:
+    2
+    3 NFC: Tag Reader
```
[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/doc_styleguide.rst]: I'll replace this with AWS FOTA.

File: doc/nrf/doc_styleguide.rst (hunk 5 above)
Comment (greg-fer) [reply doc/nrf/doc_styleguide.rst]: Right, my bad.

File: doc/nrf/doc_styleguide.rst (hunk 7 above)
Comment (greg-fer) [reply doc/nrf/doc_styleguide.rst]: Removed, thanks.

File: doc/nrf/doc_styleguide.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (ru-fu) [reply doc/nrf/doc_styleguide.rst]: I'm not talking about the "code-block:: python", but about the caption.

File: doc/nrf/doc_styleguide.rst [hunk 10]
```diff
@@ -15,40 +15,105 @@ RST style guide
...
+Use the following highlighing rules in the RST conceptual documentation:
+
+.. content_highlighting_table:
+
++------------------------+------------------------+---------------------------------------------------------+
+| Markup                 | Example                | Usage criteria                                          |
++========================+========================+=========================================================+
+| ``*emphasis*``         | *relaying*             | Emphasis or new terms.                                  |
++------------------------+------------------------+---------------------------------------------------------+
+| ````code````           | ``west update``        | Code within text, including file paths and names.       |
```
Comment (ru-fu) [inline doc/nrf/doc_styleguide.rst]: Actually, we also have the :file:`xxx` command. Do we want to establish any rules for when to use this?
I usually try to use it for file names or paths that include the file name, but not for directories. But surely not consistently.

File: doc/nrf/doc_styleguide.rst [hunk 11]
```diff
@@ -120,6 +208,41 @@ To link directly to a doxygen reference from RST, use the following Breathe doma
 
       :cpp:enumerator:`ENUM_VALUE <DOXYGEN_GROUP::ENUM_VALUE>`
 
+PCB names
+=========
+
+When referring to specific Printed Circuit Board elements, use capitalization.
+For example, **Button 1**, **Switch 3**, **LED 1**.
```
Comment (ru-fu) [inline doc/nrf/doc_styleguide.rst]: So would we write "Press Button 1 to..." or "Press **Button 1** to..."?
No big preference here (slight tendency for no bold).

File: doc/nrf/doc_styleguide.rst [hunk 12]
```diff
@@ -120,6 +208,41 @@ To link directly to a doxygen reference from RST, use the following Breathe doma
...
+
+If you are referring to a generic PCB element, do *not* use capitalization: *button*, *switch*, *LEDs*.
+
+
+Tables
+======
+
+Follow Zephyr's :ref:`zephyr:doc_guidelines` for tables.
+Do not add table titles to describe the table.
+Instead, make sure to introduce the table with a short sentence that describes table contents.
```
Comment (ru-fu) [inline doc/nrf/doc_styleguide.rst]:
```suggestion
Instead, make sure to introduce the table with a short sentence that describes the table contents.
```

File: doc/nrf/doc_styleguide.rst (hunk 10 above)
Comment (ru-fu) [reply doc/nrf/doc_styleguide.rst]: Use :file: for anything related to files - file names, file paths, directories, extensions.

File: doc/nrf/doc_styleguide.rst (hunk 11 above)
Comment (greg-fer) [reply doc/nrf/doc_styleguide.rst]: Hm, this is a bit more complicated. Rephrased in the latest version.

File: doc/nrf/doc_styleguide.rst (hunk 10 above)
Comment (greg-fer) [reply doc/nrf/doc_styleguide.rst]: Updated table as discussed during the NCS sync meeting.

**Comment by greg-fer** [comment]
@ru-fu , this now includes the edits that didn't make it to GitHub when commenting was down.

**Pull request merged**